spec/controllers/gameControllerSpec.js

```diff
@@ -2,54 +2,199 @@
+  }
+};
+
+
+function gameViewDouble(gameDouble) {
+  this.game = gameDouble;
+}
+
+gameViewDouble.prototype = {
+  clearCanvas: function() {
+  },
+  draw: function() {
+  }
+};
+
+
 describe("GameController", function() {
   beforeAll(function() {
     dummyElement = mockInterface();
-    gameController = new GameController();
+    car = new carDouble();
+    game = new gameDouble(car);
+    gameView = new gameViewDouble(game);
+    gameController = new GameController(gameView, game);
+  });
+
+  it("binds handlers to the keyup and keydown events", function() {
+    var spyAddEventListener = spyOn(window, 'addEventListener').and.callThrough();
+    gameController.bindKeys();
+    expect(spyAddEventListener).toHaveBeenCalledWith('keyup', gameController._keyupHandler, false);
+    expect(spyAddEventListener).toHaveBeenCalledWith('keydown', gameController._keydownHandler, false);
+  });
+
+  describe("spacebar has been pressed", function(){
+    var myKeyPressSpy;
+    var keyEventDown;
+
+    beforeEach(function(){
+      myKeyPressSpyDown = spyOn(gameController, 'keydown').and.callThrough();
+      myKeyPressSpyUp = spyOn(gameController, 'keyup').and.callThrough();
+      keyEventDown = mockEventKeydown(32);
+      window.dispatchEvent(keyEventDown);
+    })
+
+    it("_keydownHandler is called with a spacebar key and added to keys", function() {
+      expect(gameController.keys[32]).toBe(true);
+      expect(gameController.keys).toEqual({32 : true});
+      expect(myKeyPressSpyDown).toHaveBeenCalledWith(keyEventDown);
+    });
+
+    it("_keyupHandler is called with a spacebar key and removed from keys", function() {
+      keyEventUp = mockEventKeyup(32);
+      window.dispatchEvent(keyEventUp);
+      expect(gameController.keys[32]).toBe(undefined);
+      expect(gameController.keys).toEqual({});
+      expect(myKeyPressSpyUp).toHaveBeenCalledWith(keyEventUp);
+    });
+
+    it("accelerate was called on the car when spacebar goes up", function(){
+      var carSpy = spyOn(gameController.game.car, 'accelerate').and.callThrough();
+      gameController.countdownFinished = true;
+      keyEventUp = mockEventKeyup(32);
+      window.dispatchEvent(keyEventUp);
+      expect(carSpy).toHaveBeenCalled();
+    })
+  });
+
+  describe("enter has been pressed", function(){
+    var myKeyPressSpy;
+    var keyEventDown;
+
+    beforeEach(function(){
+      myKeyPressSpyDown = spyOn(gameController, 'keydown').and.callThrough();
+      myKeyPressSpyUp = spyOn(gameController, 'keyup').and.callThrough();
+      keyEventDown = mockEventKeydown(13);
+      window.dispatchEvent(keyEventDown);
+    });
+
+    it("_keydownHandler is called with an enter key and added to keys", function() {
+      expect(gameController.keys[13]).toBe(true);
+      expect(gameController.keys).toEqual({13 : true});
+      expect(myKeyPressSpyDown).toHaveBeenCalledWith(keyEventDown);
+    });
+
+    it("_keyupHandler is called with an enter key and removed from keys", function() {
```

Comment:
Need to test that the countdown gets started when enter key up. Also see if we can test timeout for start game.